src/utils/clients.ts

```diff
@@ -50,45 +57,50 @@ export const setConnected = (() => {
 		connected: boolean,
 		logger: Logger,
 	) => {
-		try {
-			if (uuids.length === 0) {
-				return;
-			}
-			const eventType = connected ? 'connect' : 'disconnect';
-			const response: IncomingMessage = await request
-				.post({
-					url: `https://${BALENA_API_HOST}/services/vpn/client-${eventType}`,
-					timeout: REQUEST_TIMEOUT,
-					json: true,
-					body: {
-						serviceId,
-						uuids,
-						connected,
-					},
-					headers: { Authorization: `Bearer ${apiKey}` },
-				})
-				.promise()
-				.timeout(REQUEST_TIMEOUT);
-			if (response.statusCode !== 200) {
-				throw new Error(
-					`Status code was '${response.statusCode}', expected '200'`,
-				);
-			}
-			// Update the current state on success
-			for (const uuid of uuids) {
-				deviceStates[uuid].currentConnected = connected;
-				logger.debug(
-					`successfully updated state for device: uuid=${uuid} connected=${connected}`,
-				);
-			}
+		if (uuids.length === 0) {
 			return;
-		} catch (err) {
-			captureException(err, 'device-state-update-error');
-			for (const uuid of uuids) {
-				// If we failed then add the uuids back into the list of pending updates
-				pendingUpdates.add(uuid);
-			}
 		}
+		const eventType = connected ? 'connect' : 'disconnect';
+		const uuidChucks = _.chunk(uuids, API_DEVICE_STATE_POST_BATCH_SIZE);
+		await Promise.allSettled(
+			uuidChucks.map(async (uuidChuck) => {
+				try {
+					const response: Response = await request
```

Comment:
What does typescript infer the type to be?

Reply:
It was any previously, so in guessing that the request-promise typings are not the best.